test/presenters/return-logs/setup/start.presenter.test.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = (exports.lab = Lab.script())
+const { expect } = Code
+
+// Thing under test
+const StartPresenter = require('../../../../app/presenters/return-logs/setup/start.presenter.js')
+
+describe('Return Logs Setup - Start presenter', () => {
+  let session
+
+  describe('when provided with a populated session', () => {
+    beforeEach(() => {
+      session = _testSession()
+    })
+
+    it('correctly presents the data', () => {
+      const result = StartPresenter.go(session)
+
+      expect(result).to.equal({
+        abstractionPeriod: 'From 1 January to 31 December',
+        displayRecordReceipt: true,
+        licenceId: 'db3731ae-3dde-4778-a81e-9be549cfc0e1',
+        licenceRef: '01/111',
+        pageTitle: 'Abstraction return',
+        purposes: 'Evaporative Cooling',
+        returnLogId: 'v1:6:01/111:2222:2022-04-01:2005-03-31',
+        returnsPeriod: 'From 1 April 2022 to 31 March 2023',
+        returnReference: '1234',
+        selectedOption: undefined,
+        siteDescription: 'POINT A, TIDAL RIVER MEDWAY AT ISLE OF GRAIN',
+        status: 'overdue',
+        tariffType: 'Standard tariff'
+      })
+    })
+
+    describe('the "displayRecordReceipt" property', () => {
+      describe('when the return log has a received date', () => {
+        beforeEach(() => {
+          session.receivedDate = '2023-04-27T00:00:00.000Z'
+        })
+
+        it('sets the property to false', () => {
+          const result = StartPresenter.go(session)
+
+          expect(result.displayRecordReceipt).to.be.false()
+        })
+      })
+    })
+
+    describe('the "selectedOption" property', () => {
+      describe('when an option has been selected and submitted', () => {
+        beforeEach(() => {
+          session.whatToDo = 'enterReturn'
+        })
+
+        it('returns the selected option', () => {
+          const result = StartPresenter.go(session)
+
+          expect(result.selectedOption).to.equal('enterReturn')
+        })
+      })
+    })
+
+    describe('the "status" property', () => {
+      describe('when the status is "completed"', () => {
+        beforeEach(() => {
+          session.status = 'completed'
+        })
+
+        it('returns the status as "complete', () => {
+          const result = StartPresenter.go(session)
+
+          expect(result.status).to.equal('complete')
+        })
+      })
+
+      describe('when the status is not yet "overdue"', () => {
```

Comment:
I know 'overdue' is covered by the test above. However, our unit tests can also serve as a form of documentation. So, usually, in the presenter tests, if a property is based on some logic, we try to cover _all_ expected behaviours in the tests, even if the main "correctly presents the data" already covers it.

So, I would add a final scenario to this group to document that the status will be' overdue' when the due date is less than `today()'.

> This applies to all the properties. Just noticed they are only testing behaviour not covered by "correctly presents the data".